tree/treeplayer/inc/ROOT/TDFInterface.hxx

```diff
@@ -1099,12 +1113,14 @@ protected:
                files[slot] = merger.GetFile();
                trees[slot] = new TTree(treename.c_str(), treename.c_str());
                trees[slot]->ResetBit(kMustCleanup);
+               trees[slot]->SetAutoFlush(autoflush);
                // hack to call TTree::Branch on all variadic template arguments
                std::initializer_list<int> expander = {(trees[slot]->Branch(bnames[S].c_str(), &args), 0)..., 0};
                (void)expander; // avoid unused variable warnings for older compilers such as gcc 4.9
             }
             trees[slot]->Fill();
-            if (files[slot]->GetBytesWritten() >= cachesize) files[slot]->Write();
+            if (autoflush > 0 && trees[slot]->GetEntries() % autoflush == 0)
```

Comment:
It may be better if you stop by the office because we seem to be talking past each other ...

I understand and agree with all statements you made in your last comment but still certain it has nothing to do with testing (in addition to other test) for fAutoFlush being negative ...